CPAC/utils/utils.py

```diff
@@ -1384,18 +1388,463 @@ def concat_list(in_list1=None, in_list2=None):
+         r'${functional_registration.2-func_registration_to_template.'
+         'output_resolution.func_preproc_outputs}')
+    ]
+    for keylist in [
+        ['anatomical_preproc', 'registration_workflow',
+         'template_brain_only_for_anat'],
+        ['anatomical_preproc', 'registration_workflow',
+         'template_skull_for_anat'],
+        ['anatomical_preproc', 'registration_workflow', 'registration',
+         'FSL-FNIRT', 'ref_mask'],
+        ['functional_registration', '2-func_registration_to_template',
+         'target_template', 'T1_template', 'template_brain'],
+        ['functional_registration', '2-func_registration_to_template',
+         'target_template', 'T1_template', 'template_skull'],
+        ['voxel_mirrored_homotopic_connectivity', 'symmetric_registration',
+         'template_symmetric_brain_only'],
+        ['voxel_mirrored_homotopic_connectivity', 'symmetric_registration',
+         'template_symmetric_skull'],
+        ['voxel_mirrored_homotopic_connectivity', 'symmetric_registration',
+         'dilated_symmetric_brain_mask'],
+        ['PyPEER', 'eye_mask_path']
+    ]:
+        value = lookup_nested_value(d, keylist)
+        if isinstance(value, str):
+            for replacement in resolution_replacements:
+                value = value.replace(*replacement)
+            if value:
+                set_nested_value(d, keylist, value)
+
+    bet = lookup_nested_value(d, [
+        'anatomical_preproc', 'brain_extraction', 'extraction', 'using'])
+    if bet:
+        for replacement in [
+            ('AFNI', '3dSkullStrip'), ('FSL', 'BET'), ('unet', 'UNet')
+        ]:
+            bet = list_item_replace(bet, *replacement)
+        d = set_nested_value(
+            d,
+            ['anatomical_preproc', 'brain_extraction', 'extraction', 'using'], bet)
+
+    seg_use_threshold = lookup_nested_value(d, [
+        'anatomical_preproc', 'segmentation_workflow',
+        '3-custom_thresholding', 'run'])
+    if seg_use_threshold:
+        if 'FSL-FAST Thresholding' in seg_use_threshold:
+            if 'using' in d['anatomical_preproc']['segmentation_workflow'].get(
+                '1-segmentation', {}
+            ):
+                d['anatomical_preproc']['segmentation_workflow'][
+                    '1-segmentation']['using'].append('FSL-FAST')
+            else:
+                set_nested_value(d, [
+                    'anatomical_preproc', 'segmentation_workflow',
+                    '1-segmentation', 'using'], ['FSL-FAST'])
+            seg_use_threshold.remove('FSL-FAST Thresholding')
+        if 'Customized Thresholding' in seg_use_threshold:
+            set_nested_value(d, [
+                'anatomical_preproc', 'segmentation_workflow',
+                '3-custom_thresholding', 'run'], [True])
+        else:
+            set_nested_value(d, [
+                'anatomical_preproc', 'segmentation_workflow',
+                '3-custom_thresholding', 'run'], False)
```

Comment:
@sgiavasis I'd appreciate some eyes on this change in particular. Is my reading of the change here correct?

* If `'FSL-FAST Thresholding'` was in `seg_use_threshold` in the flat config, `'FSL-FAST'` should be in `anatomical_preproc['segmentation_workflow']['1-segmentation']['using']` in the nested config
* If `'Customized Thresholding'` was in `seg_use_threshold` in the flat config, `True` should be in `anatomical_preproc['segmentation_workflow'][ '3-custom_thresholding']['run']` in the nested config

Reply:
That looks correct to me.

Reply:
It reminds me that `anatomical_preproc['segmentation_workflow'][ '3-custom_thresholding']['run']` should be `off` in default_pipeline.yml